### PR TITLE
Make the NAWRAKU scene from Reksio i Skarb Piratów playable

### PR DESCRIPTION
**Pull request by Dove6**

### CODE REVIEW

File: src/engine/types/button.ts [hunk 1]
```diff
@@ -22,10 +22,14 @@ export class Button extends Type<ButtonDefinition> {
...
 
+        this.callbacks.register('ONACTION', definition.ONACTION)
+        this.callbacks.register('ONCLICKED', definition.ONCLICKED)
+        this.callbacks.register('ONDRAGGING', definition.ONDRAGGING)
+        this.callbacks.register('ONENDDRAGGING', definition.ONENDDRAGGING)
         this.callbacks.register('ONFOCUSON', definition.ONFOCUSON)
         this.callbacks.register('ONFOCUSOFF', definition.ONFOCUSOFF)
-        this.callbacks.register('ONCLICKED', definition.ONCLICKED)
         this.callbacks.register('ONRELEASED', definition.ONRELEASED)
+        this.callbacks.register('ONSTARTDRAGGING', definition.ONSTARTDRAGGING)
```
Comment (Dove6) [inline src/engine/types/button.ts:32]: Added all signals/props found in scripts, but without implementing them rn

File: src/engine/types/index.ts [hunk 2]
```diff
@@ -60,11 +60,6 @@ export class Type<DefinitionType extends TypeDefinition> {
 export class DisplayType<DefinitionType extends DisplayTypeDefinition> extends Type<DefinitionType> {
     private priority: number = 0
 
-    constructor(engine: Engine, definition: DefinitionType) {
-        super(engine, definition)
-        this.priority = definition.PRIORITY ?? 0
-    }
-
```
Comment (Dove6) [inline src/engine/types/index.ts:67]: I hope it's not required to leave sth like xD
```
    constructor(engine: Engine, definition: DefinitionType) {
        super(engine, definition)
    }
```

Comment (Adikso) [reply src/engine/types/index.ts:67]: You removed assigning priority to this.priority, so it won't return the correct priority with GETPRIORITY?

File: src/engine/types/sequence.ts [hunk 3]
```diff
@@ -264,7 +264,7 @@ export class Sequence extends Type<SequenceDefinition> {
     }
 
     ISPLAYING() {
-        throw new NotImplementedError()
+        return this.sequenceName !== null
```
Comment (Adikso) [inline src/engine/types/sequence.ts:267]: Will sequenceName be null after sequence ends by itself? STOP() is not called when sequence ends

File: src/engine/types/button.ts [hunk 4]
```diff
@@ -63,6 +67,10 @@ export class Button extends Type<ButtonDefinition> {
     }
 
     private setRect(rect: number[] | reference) {
+        if (this.gfxStandard) {
+            this.unregisterInteractive(this.gfxStandard)
```
Comment (Adikso) [inline src/engine/types/button.ts:71]: I guess that its not possible to undo SETRECT right?

File: src/engine/types/index.ts (hunk 2 above)
Comment (Dove6) [reply src/engine/types/index.ts:67]: Hmm, that's right. Should `GETPRIORITY` be working before `init()`?

Comment (Adikso) [reply src/engine/types/index.ts:67]: It's not gonna be called before init() so no

File: src/engine/types/button.ts (hunk 4 above)
Comment (Dove6) [reply src/engine/types/button.ts:71]: That's right. Should have put a comment.

File: src/engine/types/sequence.ts (hunk 3 above)
Comment (Dove6) [reply src/engine/types/sequence.ts:268]: Good catch, fixed that in `progressNext` (L159) just now.

File: src/engine/types/index.ts (hunk 2 above)
Comment (Dove6) [reply src/engine/types/index.ts:67]: Could alternatively implement an `init()` here to be called at the end of `init()` in children.

```ts
init() {
    this.getRenderObject()!.zIndex = this.priority
    this.engine.sortObjects()
}
```

**Comment by Adikso** [comment]
What about this priority field? It seems that its not gonna work still 🤔

File: src/engine/types/index.ts (hunk 2 above)
Comment (Adikso) [reply src/engine/types/index.ts:67]: What problem would that solve? 🤔

File: src/engine/types/sequence.ts (hunk 3 above)
Comment (Adikso) [reply src/engine/types/sequence.ts:268]: This fix broke kogut in S14_0_PRZEDKOGUTEM Ufo

Comment (Dove6) [reply src/engine/types/sequence.ts:268]: ohno

Comment (Dove6) [reply src/engine/types/sequence.ts:269]: Yeah, putting it *before* ONFINISHED is definitely a better idea.

File: src/engine/types/index.ts (hunk 2 above)
Comment (Dove6) [reply src/engine/types/index.ts:67]: Code duplication (by introducing more code duplication).  
OK, I guess it's not relevant.

**Comment by Dove6** [comment]
> What about this priority field? It seems that its not gonna work still 🤔

What do you mean?

**Comment by Adikso** [comment]
> > What about this priority field? It seems that its not gonna work still 🤔
> 
> What do you mean?

Oh, it's being set, alright

**Review by Adikso — APPROVED** [review]

**Pull request merged**